src/Interpreters/AggregateDescription.h

```diff
@@ -21,4 +26,52 @@ struct AggregateDescription
 
 using AggregateDescriptions = std::vector<AggregateDescription>;
 
+
+struct WindowFunctionDescription
+{
+    // According to the standard, a window function can refer to a window declared
+    // elsewhere, using its name. We haven't implemented this yet: all windows
+    // are declared in OVER clause, but they still get an auto-generated name,
+    // and this name is used to find the window that corresponds to a function.
+    std::string window_name;
+
+    std::string column_name;
+    const ASTFunction * function_node;
+    AggregateFunctionPtr aggregate_function;
+    Array function_parameters;
+    DataTypes argument_types;
+    Names argument_names;
+
+    std::string dump() const;
+};
+
+struct WindowDescription
+{
+    std::string window_name;
+
+    // We don't care about the particular order of keys for PARTITION BY, only
+    // that they are sorted. For now we always require ASC, but we could be more
+    // flexible and match any direction, or even different order of columns.
+    SortDescription partition_by;
+
+    SortDescription order_by;
+
+    // To calculate the window function, we sort input data first by PARTITION BY,
+    // then by ORDER BY. This field holds this combined sort order.
+    SortDescription full_sort_description;
+
+    // No frame info as of yet.
+
+
+    // Reverse map to function descriptions, for convenience of building the
```

Comment:
It is not clear which map is reversed.
It is logical to have a list of window functions for a window, though.
Is order of window functions important?